packages/core/src/hooks/usePromiseTransaction.ts

```diff
@@ -26,10 +31,38 @@ export function usePromiseTransaction(chainId: number | undefined, options?: Tra
         const receipt = await transaction.wait()
         setState({ receipt, transaction, status: 'Success', chainId })
         return receipt
-      } catch (e) {
+      } catch (e: any) {
         const errorMessage = e.error?.message ?? e.reason ?? e.data?.message ?? e.message
         if (transaction) {
-          setState({ status: 'Fail', transaction, receipt: e.receipt, errorMessage, chainId })
+          const droppedAndReplaced = isDroppedAndReplaced(e)
+
+          if (droppedAndReplaced) {
```

Comment:
```suggestion
          if (e.code === errors.TRANSACTION_REPLACED) {
```

Ethers provides codes to check what type of error is thrown